static/js/util.js

```diff
@@ -205,6 +205,21 @@ exports.is_all_or_everyone_mentioned = function (message_content) {
     return all_everyone_re.test(message_content);
 };
 
+exports.is_all_mentioned = function (message_content) {
+    const all_re = /(^|\s)(@\*{2}(all)\*{2})($|\s)/;
+    return all_re.test(message_content);
+};
+
+exports.is_everyone_mentioned = function (message_content) {
+    const everyone_re = /(^|\s)(@\*{2}(everyone)\*{2})($|\s)/;
+    return everyone_re.test(message_content);
+};
+
+exports.is_stream_mentioned = function (message_content) {
+    const stream_re = /(^|\s)(@\*{2}(stream)\*{2})($|\s)/;
+    return stream_re.test(message_content);
+};
```

Comment:
Rather than adding these 3 functions, I think it'd be cleaner to just modify the interface for `exports.is_all_or_everyone_mentioned` to return the specific mention text (Or undefined if there is no wildcard mention); probably this would involve renaming it to `exports.find_wildcard_mentions`

Reply:
Thanks for this suggestion! It also helped in cleaning up the wildcard mention check inside compose.js

Reply:
This is a lot better, but I think you do even better.  If you just use the capture groups from the original regular expression, you don't need all the conditionals here -- you can just return null or the value captured by trying to match the message.

Reply:
The value captured is an array of all matches and the exact wildcard mention is its fourth element. So had to do a bit of hacky-ish code.